src/libraries/System.Runtime.InteropServices.RuntimeInformation/src/System/Runtime/InteropServices/RuntimeInformation/RuntimeInformation.Browser.cs

```diff
@@ -0,0 +1,17 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Runtime.InteropServices
+{
+    public static partial class RuntimeInformation
+    {
+        public static bool IsOSPlatform(OSPlatform osPlatform) => osPlatform.Equals(OSPlatform.Browser);
+
+        public static string OSDescription => "web";
+
+        public static Architecture OSArchitecture => Architecture.Wasm;
+
+        public static Architecture ProcessArchitecture => throw new PlatformNotSupportedException();
```

Comment:
Why does this throw?  There's no meaningful value we can return?

Reply:
yeah I'd expect this to return `Architecture.Wasm`

Reply:
It's quite odd to me to have Architecture.Wasm as ProcessArchitecture. Maybe we should have something like Architecture.Unknown.

Reply:
Wasm is functionally a processor architecture like x86 or arm. It has its own instruction set and register types

Reply:
I agree with Katelyn. Either throw PNSE from both OSArchitecture and ProcessArchitecture or use Wasm for both (I prefer the latter).

Reply:
ok, will keep it simple ;-)